### PR TITLE
Fix ie11 test

### PR DESCRIPTION
**Pull request by millerds**
This should this git the test to compile such that the ie11 test can run correctly and be re-enabled.

### CODE REVIEW

**Review by igor-ribeiiro — APPROVED** [review]

**Review by igor-ribeiiro — COMMENTED** [review]
Where did you find these find process files?

**Comment by millerds** [comment]
> Where did you find these find process files?

I looked at the code from the find-process package in github and took the portion of them that matched what we needed.  I wanted to narrow it down even more, but that would risk breaking the code that is verified working and would take more time.

File: tsconfig.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "compilerOptions": {
     "allowJs": true,
     "baseUrl": ".",
+    "esModuleInterop": true,
```
[the remark this answers is not in the document]
Comment (millerds) [reply tsconfig.json:5]: Directly for this fix no . . . it does make us more bullet proof for future changes that need to be compatibly with IE webview, but it's not needed for this change directly now that I've added the find code locally (it was part of the first change the turned out not to work).

**Comment by igor-ribeiiro** [comment]
> > Where did you find these find process files?
> 
> I looked at the code from the find-process package in github and took the portion of them that matched what we needed. I wanted to narrow it down even more, but that would risk breaking the code that is verified working and would take more time.

Why isn't the node find-process package working?

**Comment by millerds** [comment]
> > > Where did you find these find process files?
> > 
> > 
> > I looked at the code from the find-process package in github and took the portion of them that matched what we needed. I wanted to narrow it down even more, but that would risk breaking the code that is verified working and would take more time.
> 
> Why isn't the node find-process package working?

I wasn't able to find out specifically why, but it's the use of this library that cause the test in IE webview to stop working . . . the test taskpane fails to load because of a javascript error (I was unable to figure out what specific error).  It's related to trying to compile for down level es5 standard (which we've setup in the compilation process for the taskpane projects).

**Comment by igor-ribeiiro** [comment]
But if you simply copied the code from npm and put it locally, what changed that would make the local code work?

**Comment by millerds** [comment]
It's compiled from our local code rather then just importing the package.  I also tweaked how the main module is defined.  It works for the IE 11 web view (still trying to confirm mac doesn't break).

File: test/end-to-end/src/find-process-by-name.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
+/* global require, process */
+
+"use strict";
+
+const path = require("path");
+const utils = require("./find-process-utils");
```
Comment (akrantz) [inline test/end-to-end/src/find-process-by-name.ts:6]: You should use `import` and not `require()` with TypeScript.

File: test/end-to-end/src/find-process-by-name.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+function matchName(text, name) {
```
Comment (akrantz) [inline test/end-to-end/src/find-process-by-name.ts:8]: What is all this code doing? I find this hard to read and understand.

File: test/end-to-end/src/find-process-by-name.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+            cmd: row.CommandLine,
+          }));
+        resolve(list);
+      });
+    });
+  },
+};
+
+export function findProcess(cond): any {
+  const platform = process.platform;
```
Comment (akrantz) [inline test/end-to-end/src/find-process-by-name.ts:161]: Should there be separate functions for each platform? If you're writing your own code to find the process id given a process name, shouldn't this be done separately outside the test code itself? Was this an imported package that we stopped using? Why not have functionality from office-addin-debugging which already has some process stuff in pace?

**Review by akrantz — CHANGES_REQUESTED** [review]
I don't think it's a good idea to write this code in this way. It's not abstracted, organized, readable, or very maintainable. Please see other comments.

**Comment by millerds** [comment]
Managed to find a kill by process name command for the command line I can trigger on the mac.  Much simpler and more direct.

**Review by igor-ribeiiro — APPROVED** [review]

**Comment by millerds** [comment]
Discussed offline.

**Pull request merged**